### DISCUSSION TITLE
O3D312 Canno't display 2D or 3D image in Monitoring of ifm_VisionAssistant。

### DISCUSSION THREAD

**Discussion by mikufans-dumo** [ifm/ifm3d#454] [Q&A] [answered]
Please help！！！！
It‘s my first use 3D TOF camera，I don't konw how to resolve this problem. The problems are as follows: 

First:
Unable to display 2D or 3D images on the Monitoring interface after setting up using the wizard wizard:

![微信图片_20240713101029](https://github.com/user-attachments/assets/c079c3bb-ac66-45e6-bf19-b715d21a6b6d)    
(2D)
![微信图片_20240713101011](https://github.com/user-attachments/assets/2e8ba1b3-7967-4748-b352-b7f97c8a422d)
(3D)

**However,** during I setting the wizard  the image can be displayed  and  tested normally:
![微信图片_20240713100958](https://github.com/user-attachments/assets/3a817078-f60f-4917-b831-edc8ae3a95c3)
(wizard setting interface 1)

Second:
Using User defined mode, Monitoring interface can display normally without adding any templates or other settings.
![微信图片_20240713101025](https://github.com/user-attachments/assets/8608fe9e-b4ef-4cd9-a4de-deb9b7aa07ed)
(wizard setting interface 2)

**Hower**, after adding ROI distance underfill test Model or other Model, it still cannot be displayed on the Monitoring interface.
![微信图片_20240713101035](https://github.com/user-attachments/assets/9ec46956-8706-4cc5-b60a-ed6abcfe9fe2)
(wizard setting interface 3)

**ifm_VisionAssistant Vision** : ifm_VisionAssistant_V2.8.7
**O3D_Device_Details_2024-07-13_094831.txt**:
Hardware information:
Connector: #!02_A310_C41_07057235_008026112
Diagnose: #!03_D312_C34_07466109_008031205
Frontend: #!02_F342_C35_01114605_008030979
Illumination: #!02_I300_001_07440342_008001175
MACAddress: 00:02:01:44:D7:F9
Mainboard: #!03_M381_C41_07458391_008023690
MiraSerial: 9030-1b51-0259-0a24
Article: O3D312 AF

Software information:
Algorithm_Version: 2.2.14
Calibration_Device: 00:02:01:44:d7:f9
Calibration_Version: 1.0.2
Diagnostic_Controller: v1.0.44-2dcaabbc73-dirty
IFM_Recovery: O3D3XX 0.6.73
IFM_Software: 1.30.5309
Linux: Linux version 3.14.34-rt31-yocto-standard-00048-gc8112a91aeb2-dirty (jenkins@dettlx190) (gcc version 4.9.2 (GCC) ) #1 SMP PREEMPT RT Wed Mar 27 18:25:10 CET 2019
Main_Application: 1.0.44

Application List:
1: [1484739682] Completeness application
2: [861609334] 完整性应用
3: [984411720] Mytest
4: [631185153] Myapplication
5: [476707713] Completeness application
6: [1571000710] Advanced application
7: [1484739683] Completeness application
8> [372314962] Completeness application

Logging information:
T: 9.8284, P: 417,  C: WARN,  S: iom,  M: Program starting ...
T: 9.8304, P: 414,  C: INFO,  S: qt_dbus_web_bridge,  M: qt_dbus_web_bridge: Enabled Logging via Redis
T: 9.8331, P: 414,  C: WARN,  S: qt_dbus_web_bridge,  M: Starting ...
T: 9.9560, P: 417,  C: INFO,  S: iom,  M: DIAGC_DBUS_EVENT_DIAGC_BOOTED
T: 9.9620, P: 417,  C: INFO,  S: diagCtrl,  M: sysboot_hw_evt[132]: successfully init HW : ###
T: 10.6271, P: 453,  C: WARN,  S: suprge,  M: suprge: supervision activity starting, Bello the watchdog, is unfused ...
T: 10.7327, P: 453,  C: WARN,  S: suprge,  M: Registerinng services for the O3D3XX family of devices
T: 11.0988, P: 443,  C: INFO,  S: gecco,  M: gecco: Enabled Logging via Redis
T: 11.1002, P: 443,  C: WARN,  S: gecco,  M: Starting ...
T: 11.9808, P: 431,  C: INFO,  S: GECOM,  M: DBUS connection successful
T: 12.3903, P: 439,  C: INFO,  S: imager,  M: imager: Enabled Logging via Redis
T: 12.3909, P: 439,  C: WARN,  S: imager,  M: Starting ...
T: 13.4687, P: 443,  C: ERROR,  S: gecco,  M: bool ifm::O3D3xxDevice::activateApplication(): application activation failed, probably no imager process running
T: 13.4698, P: 443,  C: WARN,  S: gecco,  M: QDBusConnection: name 'com.ifm.suprge' had owner '' but we thought it was ':1.5'
T: 13.8305, P: 417,  C: INFO,  S: diagCtrl,  M: Stop VLED monitoring #
T: 13.8661, P: 439,  C: WARN,  S: imager,  M: void gelfi::HawkeyeStepperMotor::initSystem(bool, const gelfi::HawkeyeStepperMotor*): No motor is available for this article
T: 14.4222, P: 417,  C: INFO,  S: diagCtrl,  M: set_illu_power 1 #
T: 15.2986, P: 417,  C: INFO,  S: diagCtrl,  M: Start VLED monitoring #
T: 15.3012, P: 443,  C: WARN,  S: gecco,  M: Imager is available again
T: 15.3026, P: 439,  C: WARN,  S: imager,  M: LRAE is available again
T: 17.0578, P: 443,  C: WARN,  S: gecco,  M: Finished imager configuration
T: 1106.6699, P: 439,  C: ERROR,  S: imager,  M: Persistent data invalid - new teach is required
T: 1147.6782, P: 439,  C: ERROR,  S: imager,  M: Persistent data invalid - new teach is required
T: 1160.3098, P: 439,  C: ERROR,  S: imager,  M: Persistent data invalid - new teach is required
T: 1169.2902, P: 439,  C: ERROR,  S: imager,  M: Persistent data invalid - new teach is required
T: 2793.8926, P: 439,  C: ERROR,  S: imager,  M: Persistent data invalid - new teach is required
T: 2835.6458, P: 439,  C: ERROR,  S: imager,  M: Persistent data invalid - new teach is required
T: 2840.9260, P: 439,  C: ERROR,  S: imager,  M: Persistent data invalid - new teach is required

**image setting:**
AutoExposureMaxExposureTime: 10000
AutoExposureMaxLongShortFactor: 50
AutoExposureMinLongShortFactor: 2
AutoExposureReferencePointX: 88
AutoExposureReferencePointY: 66
AutoExposureReferenceROI: {"ROIs":[{"id":0,"group":0,"type":"Rect","width":130,"height":100,"angle":0,"center_x":88,"center_y":66}]}
AutoExposureReferenceType: 0
Channel: 0
ClippingBottom: 131
ClippingCuboid: {"XMin": -3.402823e+38, "XMax": 3.402823e+38, "YMin": -3.402823e+38, "YMax": 3.402823e+38, "ZMin": -3.402823e+38, "ZMax": 3.402823e+38}
ClippingLeft: 0
ClippingRight: 175
ClippingTop: 0
ContinuousAutoExposure: false
ContinuousUserFrameCalibration: false
EnableAmplitudeCorrection: true
EnableFastFrequency: false
EnableFilterAmplitudeImage: true
EnableFilterDistanceImage: true
EnableRectificationAmplitudeImage: true
EnableRectificationDistanceImage: true
ExposureTime: 1320
ExposureTimeList: 40;1320
ExposureTimeRatio: 33
FlyingPixelEraser: 0
FrameRate: 3
MaxAllowedLEDFrameRate: 77.599999999999994
MinimumAmplitude: 42
Resolution: 0
SpatialFilterType: 0
SymmetryThreshold: 0.40000000000000002
TemporalFilterType: 0
ThreeFreqMax2FLineDistPercentage: 80
ThreeFreqMax3FLineDistPercentage: 80
TotalExposureTime: 6440
TwoFreqMaxLineDistPercentage: 80
Type: under5m_moderate
UseSimpleBinning: false

**Comment by dekhanra** [comment, marked as answer]
Hello @mikufans-dumo,

Could you let us know the Trigger Source you have selected in the application? 

1. Click on Application window
2. Click on the desired application and ifmVisionAssistant will prompt 'stop the sensor' --> Click Yes and Continue
![image](https://github.com/user-attachments/assets/ec69bf36-1c07-4f3e-b465-5b32cc798958)

3. Click on the 'Edit Application'
![image](https://github.com/user-attachments/assets/441cf0ec-fee9-4831-bea3-77f13ad48241)

4. Select the Trigger Source to receive the data

![image](https://github.com/user-attachments/assets/9c6adfda-18cb-43e7-8561-2a2777260a97)


And for the further debugging, please provide us always with the configuration of the device. Go to 'Device Setup'
 and click on Export Settings and send us the file.
 
![image](https://github.com/user-attachments/assets/ddab239a-94c8-4097-8583-650dc37a8535)

**Reply by mikufans-dumo** [reply]
Thanks！！！！It worked！
